### DISCUSSION TITLE
Permissions GeoNature

### DISCUSSION THREAD

**Discussion by coder-pour-changer-de-vie** [PnX-SI/GeoNature#2808] [Q&A] [answered]
Bonjour à tous,

J'ai lu avec attention la documentation concernant la gestion des permissions dans GeoNature, mais je rencontre un problème dont la solution est peut être dans l'historique du projet.

### CONTEXTE
Je travaille acutellement au développement d'un module externe sur la base de la version 2.11.0 de GeoNature (car version en production).

Comme conseillé, je m'appuie sur le projet [cookiecutter-generation](https://github.com/PnX-SI/gn_module_template/tree/cookiecutter-generation) afin d'aligner la structure du module avec ce qui est attendu par GeoNature 

Je prend également en exemple le [module import](https://github.com/PnX-SI/gn_module_import).

Dans l'UI de GeoNature, je vois bien l'entrée de mon module (Projets) dans la barre de navigation sur la gauche, quand je clique dessus un espace vide dédié au module s'affiche.

Un texte présent dans le composant Angular principale devrait s'afficher, mais ce n'est pas le cas, et je pense que cela est lié aux permissions du module à définir.

![Capture d’écran du 2023-11-28 10-08-08](https://github.com/PnX-SI/GeoNature/assets/46313992/346a1e0d-b924-47c3-b733-7c5ae5284783)

### La problématique
Pour la première itération du module, j'ai décidé de passer en direct pour l'initialisation SQL (sans passer par Alembic), j'ai donc trouvé le code nécessaire à la définition des permissions dans le projet "cookiecutter-generation".

Cependant, l'instance de développement (fonctionnelle), ne comporte pas plusieurs tables nécessaires dans le schema : 

![Capture d’écran du 2023-11-28 10-09-34](https://github.com/PnX-SI/GeoNature/assets/46313992/0e452403-8394-4d2c-bc12-f118bb70888b)

Ils semble manquer les tables suivantes : 

- gn_permissions.t_permissions_available
- gn_permissions.t_permissions

### Questions
D'après vous : 

-  est-ce lié à la version 2.11.0 qui utilisait un mode de gestion de permission différent ?
-  cela peut-il venir d'un problème d'installation / déploiement (pourtant le front + back semblent fonctionnels) ?
-  pensez-vous que l'affichage "vide" du module est lié uniquement à la gestion des permissions du module ?

Je cherche une piste, d'avant merci pour votre aide.

Nicolas.

**Comment by jacquesfize** [comment]
Bonjour,

Bonjour, 

Le template cookie-cutter s'appuie sur la dernière version de GeoNature (2.13.x). Depuis la version 2.13, le système de permissions a complètement était revu (voir [Changelog](https://github.com/PnX-SI/GeoNature/releases/tag/2.13.0) de la 2.13). En 2.11, les permissions n'était pas dans `t_permissions` et `t_permissions_available`. Ces dernières sont découvertes "automatiquement"  et éditables dans le panneau admin dédié (accessible au travers de l'onglet "Admin").

**P.S.** Si vous voulez utiliser le template et si c'est possible, je ne peux que vous conseiller de passer en 2.13.

**Reply by coder-pour-changer-de-vie** [reply]
Je comprends, et surtout merci !

Une étude du chantier de migration est en cours.

Du coup il n'y aurait rien à faire en version 2.11.0 pour que le module s'affiche si on se connecte en tant qu'administrateur (attribution des permissions via héritage si je comprends bien) ?

Dans ce cas je ne sais pas pourquoi le contenu du module n'affiche rien :-(

Si vous aviez une piste, je continue mes tests pour comprendre

**Reply by jacquesfize** [reply]
Par défaut, un administrateur a tous les accès sur Geonature (dans la 2.11). Normalement, cela se vérifie sur la page de gestion des permissions.

Sur le pourquoi rien ne s'affiche, vous avez regardé dans le log côté frontend ? (navigateur et angular)

**Comment by camillemonchicourt** [comment, marked as answer]
En effet, il est fortement conseillé que vous réalisiez vos développements sur une version à jour de GeoNature, car il y a eu des changements importants depuis la 2.11 et c'est compliqué pour nous d'aider sur des anciennes versions.
De puis la doc, le template de module, etc... sont maintenus et suivis par rapport à la dernière version de GeoNature.

Sinon, pour les modules, après installation, il doit être listé dans la table `gn_commons.t_modules`, ainsi qu'un lien symbolique vers le dossier du module, dans le dossier `external_modules` de GeoNature.
Tout ça est géré automatiquement par la commande d'installation d'un module GeoNature.

**Reply by coder-pour-changer-de-vie** [reply]
Oui merci beaucoup, nous lançons un projet de migration pour la production.

C'est déjà très appréciable d'avoir pu avoir cette aide, pour savoir si le cheminement est correct, et bénéficier de l'historique du projet.

Oui pour le gn_commons.t_modules ainsi que le ln sur external_modules, c'était validé.

Comme le suggère @jacquesfize je pense que le problème est côté front, j'ai du zappé une étape d'alignement dans le projet Angular.

Merci beaucoup, je reprendrai sur le sujet jeudi matin, c'est très appréciable d'avoir pu être guidé sur les premiers pas !

L'avantage de descendre dans les détails d'implémentation, c'est qu'on monte en compétence plus rapidement ;-)

Bonne soirée à tous !
Nicolas.
